clippy_lints/src/casts/mod.rs

```diff
@@ -657,6 +658,31 @@ declare_clippy_lint! {
     "casting a known floating-point NaN into an integer"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for raw pointers that point to temporary values.
+    ///
+    /// ### Why is this bad?
+    /// Usage of such a pointer will result in Undefined Behavior, as the pointer will stop pointing
+    /// to valid stack memory once the temporary is dropped.
+    ///
+    /// ### Example
+    /// ```rust,ignore
+    /// let p = &0u32 as *const u32;
+    /// unsafe { p.read() }; // ⚠️
```

Comment:
This isn't actually unsafe. `&0u32` creates a `&'static u32`.

Reply:
huh, is that actually the case? Weird, perhaps it would need a more complex example then

Reply:
More generally, `let x = &e` will extend the lifetime of `e` the same as if it had first been bound to a local.

Reply:
I've created a better, actually correct example.